From b5794155ef02e633a16581397facd599d3190728 Mon Sep 17 00:00:00 2001
From: Daniel Stenberg <daniel@haxx.se>
Date: Wed, 4 Dec 2013 22:46:49 +0100
Subject: [PATCH] Curl_is_connected: use proxy name in error message when proxy is used

(bug introduced in 255826c4, never present in a release)

Reported-by: Dima Tisnek
Bug: http://curl.haxx.se/mail/lib-2013-12/0006.html

[upstream commit 1cf71bd76e4a330e5b7824014c2605e4bfe1a0a5]

Signed-off-by: Kamil Dudka <kdudka@redhat.com>
---
 lib/connect.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/lib/connect.c b/lib/connect.c
index 2b5719d..413c66e 100644
--- a/lib/connect.c
+++ b/lib/connect.c
@@ -821,7 +821,8 @@ CURLcode Curl_is_connected(struct connectdata *conn,
     error = SOCKERRNO;
     data->state.os_errno = error;
     failf(data, "Failed connect to %s:%ld; %s",
-          conn->host.name, conn->port, Curl_strerror(conn, error));
+          conn->bits.proxy?conn->proxy.name:conn->host.name,
+          conn->port, Curl_strerror(conn, error));
   }
 
   return code;
-- 
1.7.1

